QCI becomes the authoritative CI registry (#454)
openshift/ci-docs: content/en/docs/how-tos/use-registries-in-build-farm.md

---

# Summary of Available Registries
All the CI images used by OpenShift CI are stored in the repository in quay.io: `quay.io/openshift/ci`, called _QCI_ for short.

> It is the authoritative CI registry and the source of truth of all CI images.

Reviewer: ```suggestion
It is the authoritative CI registry and the source of truth for all CI images.
```

{{< /rawhtml >}}

{{< alert title="Warning" color="warning" >}}
The registry `registry.svc.ci.openshift.org` has been decommissioned. If there is any reference to its image, e.g., in your Dockerfile,
please use the corresponding image in QCI instead.
{{< /alert >}}

# Container Image Data Flows

Today, two major data flows exist for container images in the OpenShift CI ecosystem. First, when a job executes on
one of the build farm clusters, container images that need to be [built](/docs/architecture/ci-operator/#building-container-images)
for the execution will exist only on that cluster. Second, when changes are merged to repositories under test, updated

> images are built on a build farm and [promoted](/docs/architecture/ci-operator/#publishing-container-images) to QCI. Users should always pull from this registry for any images they interact with. When an image changes on QCI, that change is propagated to all build farm clusters as the images there are imported from QCI when the job is executed and thus the

Reviewer: ```suggestion
images are built on a build farm and [promoted](/docs/architecture/ci-operator/#publishing-container-images) to QCI. Users should always pull any images they interact with from this registry. When an image changes on QCI, that change is propagated to all build farm clusters as the images there are imported from QCI when the job is executed and thus the
```

copies they hold are up-to-date and jobs that run there run with the correct container image versions.

{{< alert title="Info" color="info" >}}
In addition to QCI, `ci-operator` promotes images to `app.ci`'s registry `registry.ci.openshift.org` _only_ for the sake of internal automation such as Release Controllers, ART, and [mirroring CI images to an external repository](/docs/how-tos/mirroring-to-quay/#mirror-images-with-wildcard).

> Clients from CI users or their integration should _not_ reference the images there. Otherwise, please replace them with the corresponding image in QCI.

Reviewer: ```suggestion
Clients from CI users or their integration should _not_ reference the images there. Instead, please replace them with the corresponding image in QCI.
```

{{< /alert >}}

# Common Questions

> ## How do I access to QCI?

Reviewer: ```suggestion
## How do I gain access to QCI?
```

The access control to the images in QCI is delegated to the [RBACs](/docs/how-tos/rbac/) on `app.ci`.

> This is to reduce the effort on managing users in different places.

Reviewer: ```suggestion
This is done to reduce the effort of managing users in different places.
```

    namespace: my-project
```

After the pull request is merged, the manifests will be automatically applied to `app.ci`.
With the `admin_manifest.yaml` described above, the members of group `my-project-admins`
can [create bound tokens for the service account](https://docs.openshift.com/container-platform/4.13/authentication/bound-service-account-tokens.html#bound-sa-tokens-configuring-externally_bound-service-account-tokens) `image-puller`.
Once `oc` logs in to `app.ci` with the service account token, we may use the same commands above to pull the images.

> Note that we should avoid creating long-lived tokens for service accounts for a lower risk as described in [the Kubernetes document](https://kubernetes.io/docs/reference/access-authn-authz/service-accounts-admin/#create-token). Regardless of the token types, the members of group `my-project-admins`

Reviewer: ```suggestion
Note that we should avoid creating long-lived tokens for service accounts to lower security risk as described in [the Kubernetes document](https://kubernetes.io/docs/reference/access-authn-authz/service-accounts-admin/#create-token). Regardless of the token types, the members of group `my-project-admins`
```

## How do I access the latest published images for my component?

If the `ci-operator` configuration for your component configures image [`promotion`](/docs/architecture/ci-operator/#publishing-container-images),
output container images will be published to the central CI registry when changes are merged to your repository. Two main
output container images will be published to QCI when changes are merged to your repository. Two main
configurations are possible for promotion: configuring an `ImageStream` name and namespace or a namespace and a target tag.

> ### Granting Pull Privileges

Reviewer: We don't need this section anymore because of the new ServiceAccount token step, right?

Author: Exactly. Technically, the method in this section still works and we do not block the users from using it.
But we do not encourage them to use it as the created token is more desired as the users manages its lifecycle and rotation.
